### PR TITLE
feat: bytea support

### PR DESCRIPTION
**Pull request by brudil**
Lightweight support for `bytea`.

Knex returns Buffers which are a superset of Uint8Array. joist-codegen types the primitive as Uint8Array - but this should be configurable in userland with the `type` or/and CustomSerde options, if they wish to opt in to the Buffer typing. Alternatively an instanceof check can be used.

I haven't touched too much outside of `mapSimpleDbTypeToTypescriptType` for this. Followed the bigint tests within Author.

Haven't really touched filtering here - not even sure how sensible that would be to do anyway.

### CODE REVIEW

File: packages/tests/integration/migrations/1580658856631_author.ts [hunk 1]
```diff
@@ -139,6 +139,8 @@ export function up(b: MigrationBuilder): void {
     tags_of_all_books: { type: "varchar", notNull: false },
     // for testing full-text-search fields that want to use `id`
     search: { type: "text", notNull: false },
+    // for testing full-text-search fields that want to use `id`
```
Comment (stephenh) [inline packages/tests/integration/migrations/1580658856631_author.ts:142]: Looks like a stale comment?

**Review by stephenh — APPROVED** [review]
Looks great!

File: packages/tests/integration/migrations/1580658856631_author.ts (hunk 1 above)
Comment (brudil) [reply packages/tests/integration/migrations/1580658856631_author.ts]: Good spot! fixed in last commit.

**Pull request merged**